### PR TITLE
`AggTags` now has `filter()` method + unit tests

### PR DESCRIPTION
**Pull request by sseyler**
Added `filter()` method to `AggTags`. Opted to leave out fuzzy matching for the time being due to ambiguities in possible implementations. Unit tests were added for both `AggTags.fuzzy()` (originally intended to include fuzzy matching in `filter()` and `AggTags.filter()`).

### CODE REVIEW

**Comment by sseyler** [comment]
**Note**: as seen in the commits, I've moved the tests in `test_tags_fuzzy()` to a separate branch for the finishing the rest of the `AggTags` unit tests. This is for the purpose of separating the work done on `AggTags.filter()` in the PR from the rest of the work on the `AggTags` tests.

**Comment by sseyler** [comment]
**Note**: As I did above for `test_tags_fuzzy()` (and as seen in the above commit), I've moved the tests in `test_tags_getitem()` to a separate branch (aggtags_tests) for the finishing the rest of the `AggTags` unit tests.

File: src/datreant/core/agglimbs.py [hunk 1]
```diff
@@ -278,6 +278,22 @@ def fuzzy(self, tag, threshold=80, scope='all'):
 
         return tuple(matches)
 
+    def filter(self, tag):
+        """Given tag(s), get a Bundle of Treants whose tags match it.
+
+        Parameters
+        ----------
+        tag : str or list
+            Tag or tags to get fuzzy matches for.
```
Comment (dotsdl) [inline src/datreant/core/agglimbs.py]: Not doing fuzzy matching here

File: src/datreant/core/agglimbs.py [hunk 2]
```diff
@@ -278,6 +278,22 @@ def fuzzy(self, tag, threshold=80, scope='all'):
 
         return tuple(matches)
 
+    def filter(self, tag):
+        """Given tag(s), get a Bundle of Treants whose tags match it.
```
Comment (dotsdl) [inline src/datreant/core/agglimbs.py]: Probably should be written as, "Filter out a Bundle of members matching the given tag expression."

File: src/datreant/core/agglimbs.py [hunk 3]
```diff
@@ -278,6 +278,22 @@ def fuzzy(self, tag, threshold=80, scope='all'):
...
+
+        Parameters
+        ----------
+        tag : str or list
+            Tag or tags to get fuzzy matches for.
+
+        Returns
+        -------
+        Bundle
+            Bundle of Treants with matching tag(s).
```
Comment (dotsdl) [inline src/datreant/core/agglimbs.py]: "matching the given tag expression"

**Comment by dotsdl** [comment]
Very nice! Just a few comments on docs, and then it looks good to merge.

**Comment by dotsdl** [comment]
Looks good. Merging!

**Pull request merged**